### DISCUSSION TITLE
View with union query

### DISCUSSION THREAD

**Discussion by Foxsterc** [simolus3/drift#2801] [Q&A] [answered]
Hi,

Is is possible to create a view containing a union query?
Thanks,

**Comment by simolus3** [comment, marked as answer]
Union queries are not supported in the Dart API, but it is possible to define views with unions in drift files.

**Reply by Foxsterc** [reply]
Thanks. How would you apply that in a migration  so as not to affect older versions?

**Reply by simolus3** [reply]
Views are stateless (they're running the underlying query each time you run a `SELECT` on the view), so  you only need to ensure the definition of the view in the schema stays up to date. [recreateAllViews](https://drift.simonbinder.eu/api/drift/migrator/recreateallviews) is a migration API doing just that.

**Reply by Foxsterc** [reply]
Perfect, thanks!
